zkevm-circuits/src/evm_circuit/execution/error_code_store.rs

```diff
@@ -0,0 +1,311 @@
+            vec![1.expr(), 2.expr()],
+        );
+
+        let common_error_gadget = CommonErrorGadget::construct_with_return_data(
+            cb,
+            opcode.expr(),
+            4.expr(),
+            memory_address.offset(),
+            memory_address.length(),
+        );
+
+        Self {
+            opcode,
+            memory_address,
+            code_store_gas_insufficient,
+            max_code_size_exceed,
+            common_error_gadget,
+        }
+    }
+
+    fn assign_exec_step(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        _tx: &Transaction,
+        call: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        let opcode = step.opcode().unwrap();
+        self.opcode
+            .assign(region, offset, Value::known(F::from(opcode.as_u64())))?;
+        // block.get_rws(step, index).stack_value());
+        let [memory_offset, length] = [0, 1].map(|i| block.get_rws(step, i).stack_value());
+        self.memory_address
+            .assign(region, offset, memory_offset, length)?;
+
+        self.code_store_gas_insufficient.assign(
+            region,
+            offset,
+            F::from(step.gas_left),
+            F::from(GasCost::CODE_DEPOSIT_BYTE_COST * length.as_u64()),
+        )?;
+
+        self.max_code_size_exceed.assign(
+            region,
+            offset,
+            F::from(MAXCODESIZE),
+            F::from(length.as_u64()),
+        )?;
+
+        self.common_error_gadget
+            .assign(region, offset, block, call, step, 4)?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use bus_mapping::circuit_input_builder::FixedCParams;
+    use eth_types::{
+        address, bytecode, evm_types::OpcodeId, geth_types::Account, Address, Bytecode, Word,
+    };
+
+    use lazy_static::lazy_static;
+    use mock::{eth, TestContext, MOCK_ACCOUNTS};
+
+    use crate::test_util::CircuitTestBuilder;
+
+    lazy_static! {
+        static ref CALLER_ADDRESS: Address = address!("0x00bbccddee000000000000000000000000002400");
+    }
+
+    const MAXCODESIZE: u64 = 0x6000u64;
+
+    fn run_test_circuits(ctx: TestContext<2, 1>) {
+        CircuitTestBuilder::new_from_test_ctx(ctx)
+            .params(FixedCParams {
+                max_rws: 4500,
+                ..FixedCParams::default()
+            })
+            .run();
+    }
+
+    fn initialization_bytecode(is_oog: bool) -> Bytecode {
+        let memory_bytes = [0x60; 10];
+        let memory_value = Word::from_big_endian(&memory_bytes);
+        let code_len = if is_oog { 0 } else { MAXCODESIZE + 1 };
+
+        let mut code = bytecode! {
+            PUSH10(memory_value)
+            PUSH32(code_len)
+            MSTORE
+            PUSH2(if is_oog { 5 } else { code_len}) // length to copy
+            PUSH2(32u64 - u64::try_from(memory_bytes.len()).unwrap()) // offset
+            //PUSH2(0x00) // offset
```

Comment:
a leftover?

Reply:
removed in [0a6cebe](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1545/commits/0a6cebe3552cc37998f8aba7cfa1eff0014f3509) thanks.